lib/oli_web/controllers/ingest_controller.ex

```diff
@@ -0,0 +1,23 @@
+defmodule OliWeb.IngestController do
+  use OliWeb, :controller
+
+  @spec index(Plug.Conn.t(), any) :: Plug.Conn.t()
+  def index(conn, _params) do
+    render(conn, "index.html", title: "Ingest")
+  end
+
+  def upload(conn, %{"upload" => upload}) do
+
+    author = conn.assigns[:current_author]
+
+    path_upload = upload["digest"]
+
+    case Oli.Authoring.Ingest.ingest(path_upload.path, author) do
+      {:ok, project} -> redirect(conn, to: Routes.project_path(conn, :overview, project))
+      e -> render(conn, "error.html", title: "Ingest", error: e)
```

Comment:
based on the documentation for ingest, I would expect this error case to be {:error, e}. If that is true, then error.html template will render that struct instead of the inner error message